File: frontend/src/types/appointment.ts
export interface AppointmentDto {
  id: string;
  clientId: string;
  scheduledDate: string;
  completedDate?: string;
  status: AppointmentStatusDto;
  notes?: string;
  totalPrice: number;
  createdAt: string;
  updatedAt?: string;
  services: AppointmentServiceDto[];
}

export interface AppointmentServiceDto {
  serviceId: string;
  serviceName: string;
  price: number;
  quantity: number;
}

export interface CreateAppointmentDto {
  clientId: string;
  scheduledDate: string;
  notes?: string;
  services: AppointmentServiceDto[];
}

export interface UpdateAppointmentDto {
  scheduledDate: string;
  status: AppointmentStatusDto;
  notes?: string;
}

export enum AppointmentStatusDto {
  Scheduled = 1,
  InProgress = 2,
  Completed = 3,
  Cancelled = 4,
}

File: backend/CarCareSystem.Domain/Entities/Appointment.cs
namespace CarCareSystem.Domain.Entities;

public class Appointment
{
    public Guid Id { get; set; }
    public Guid ClientId { get; set; }
    public DateTime ScheduledDate { get; set; }
    public DateTime? CompletedDate { get; set; }
    public AppointmentStatus Status { get; set; }
    public string? Notes { get; set; }
    public decimal TotalPrice { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    // Navigation properties
    public Client Client { get; set; } = null!;
    public ICollection<AppointmentService> AppointmentServices { get; set; } = new List<AppointmentService>();
}

public enum AppointmentStatus
{
    Scheduled = 1,
    InProgress = 2,
    Completed = 3,
    Cancelled = 4
}
